cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileProvider.scala

```diff
@@ -36,30 +44,85 @@ class DrsCloudNioFileProvider(config: Config, scheme: String, drsPathResolver: D
   }
 
 
+  private def extractUrlForScheme(drsPath: String, urlArray: Array[Url], scheme: String): String = {
+    val schemeUrlOption = urlArray.find(u => u.url.startsWith(scheme))
+
+    schemeUrlOption match {
+      case Some(schemeUrl) => schemeUrl.url
+      case None => throw UrlNotFoundException(drsPath, scheme)
+    }
+  }
+
+
+  private def gcsInputStream(bucket: String, fileToBeLocalized: String, serviceAccount: String): ReadChannel = {
+    val credentials = GoogleCredentials.fromStream(new ByteArrayInputStream(serviceAccount.getBytes()))
```

Comment:
let's put these 2 lines at the end of the world.  I'm submitting a PR with some suggestions